### DISCUSSION TITLE
Multiplication problem

### DISCUSSION THREAD

**Discussion by Mejla0101** [nanoframework/Home#1281] [General]
Hi all,
can someone explain to me how precision can be lost when multiplying this example float x = 0.01 * 0.01;
The result is zero.  Am I completely stupid or is this a bug? Nanoframework ported to ESP32. Thanks for the reply

**Comment by alberk8** [comment]
This is dotnet,   try float x - 0.01f * 0.01f;

**Reply by Mejla0101** [reply]
Thanks for the reply, I tried that of course. The result is the same

**Reply by josesimoes** [reply]
I'm pretty sure we have a unit test covering this one...

When you say it's 0 is that because of what? ToString output? Inspecting the value?...

**Reply by Mejla0101** [reply]
Thanks for the reply,

so it looks like a ToString() error

if ((0.01f * 0.01f) == 0.0001f) Debug.WriteLine("true");
else Debug.WriteLine("false");

Result :  true

Debug.WriteLine((0.01f * 0.01f).ToString("f8"));

Result : 0.00000000

**Reply by alberk8** [reply]
```
 float x = 0.01f * 0.01f;
 Console.WriteLine($"Result: {x}  DataType: {x.GetType()}");
```
Result: 0  DataType: System.Single

My testing shows that if one side of the input, left of right is less than One (1) then the result is zero

**Reply by josesimoes** [reply]
Yes, as I was suspecting, this looks like an issue with `ToString()`. 
Converting this to an issue.

@Mejla0101 you can track the progress of this.
